tests/models/qwen2_vl/test_processing_qwen2_vl.py

```diff
@@ -0,0 +1,269 @@
+
+        # test if it raises when no input is passed
+        with pytest.raises(ValueError):
+            processor()
+
+        # test if it raises when no text is passed
+        with pytest.raises(TypeError):
+            processor(images=image_input)
+
+    def test_model_input_names(self):
+        image_processor = self.get_image_processor()
+        tokenizer = self.get_tokenizer()
+
+        processor = Qwen2VLProcessor(tokenizer=tokenizer, image_processor=image_processor)
+
+        input_str = "lower newer"
+        image_input = self.prepare_image_inputs()
+        video_inputs = self.prepare_video_inputs()
+
+        inputs = processor(text=input_str, images=image_input, videos=video_inputs)
+
+        self.assertListEqual(list(inputs.keys()), processor.model_input_names)
+
+    def test_image_processor_defaults_preserved_by_image_kwargs(self):
+        image_processor = self.get_component("image_processor", size=(234, 234))
+        tokenizer = self.get_component("tokenizer", max_length=117, padding="max_length")
+
+        processor = self.processor_class(tokenizer=tokenizer, image_processor=image_processor)
+        self.skip_processor_without_typed_kwargs(processor)
+
+        input_str = "lower newer"
+        image_input = self.prepare_image_inputs()
+
+        inputs = processor(text=input_str, images=image_input)
+        self.assertEqual(inputs["pixel_values"][0].shape[0], 1176)
+
+    def test_kwargs_overrides_default_image_processor_kwargs(self):
+        if "image_processor" not in self.processor_class.attributes:
+            self.skipTest(f"image_processor attribute not present in {self.processor_class}")
+        image_processor = self.get_component("image_processor", size=(234, 234))
+        tokenizer = self.get_component("tokenizer", max_length=117, padding="max_length")
+
+        processor = self.processor_class(tokenizer=tokenizer, image_processor=image_processor)
+        self.skip_processor_without_typed_kwargs(processor)
+
+        input_str = "lower newer"
+        image_input = self.prepare_image_inputs()
+
+        inputs = processor(text=input_str, images=image_input, size=[224, 224])
+        self.assertEqual(inputs["pixel_values"][0].shape[0], 1176)
+
+    def test_structured_kwargs_nested(self):
+        image_processor = self.get_component("image_processor")
+        tokenizer = self.get_component("tokenizer")
+
+        processor = self.processor_class(tokenizer=tokenizer, image_processor=image_processor)
+        self.skip_processor_without_typed_kwargs(processor)
+
+        input_str = "lower newer"
+        image_input = self.prepare_image_inputs()
+        video_input = self.prepare_video_inputs()
+
+        # Define the kwargs for each modality
+        all_kwargs = {
+            "common_kwargs": {"return_tensors": "pt"},
+            "images_kwargs": {"size": {"height": 214, "width": 214}},
+            "videos_kwargs": {"size": {"height": 214, "width": 214}},
```

Comment:
I'm guessing the processing doesn't work if the video and image have different image sizes set as we can't batch? 

Could we test passing one and not the other?

Reply:
yes, passing only one worked for me but I'll add better tests also

Reply:
Oke, I found that the `size` wasn't being used at all, my bad for missing that when reviewing. Since Qwen2-VL can't resize to the given size (it needs to follow some methods to smart resize and keep as much resolution as possible), the size dict will be different from other models. Added that to the docsting

Also modified all tests to follow the new `size` format and actually use if if users pass it with correct dict-keys. WDYT? Should we also find a way to support tuple-size? I'll also update model doc page soon

Reply:
> Should we also find a way to support tuple-size? 

No. Previously `size` was a tuple and this caused issues as: 
* some models had it stored as (width, height) 
* some models just stored an int. Within that set of models, some would then resize to `(int, int)` others to `(int, int * height / width)` etc. 
* some models used a tuple to denote the longest and shortest edges

There was a huge level of ambiguity and inconsistency across the configs which made image processor behaviour difficult to predict without having to dig into the code. 

Technically, tuples are supported using `get_size_dict`, but this is for backwards compatibility rather than an encouraged way to pass in inputs. 

Adding `min_pixels` and `max_pixels` isn't a light decision, as this is something which sets in stone how this will be expressed for all configs going forward. I think this change is OK but we should be cautious about just adding keys to size dict. You'll also need to update some other code which verifies the keys in the size dicts -- namely `VALID_SIZE_DICT_KEYS`

Reply:
> No. Previously size was a tuple and this caused issues as:

I see, that makes it easier in general to use max-min pixels now cause I didn't want to add the new possible keys to `VALID_SIZE_DICT_KEYS`. Qwen is and will prob be the only one accepting such kwargs. But then we couldn't call `get_size_dict` to validate the size dict is passed correctly

Oke, now I added it in VALID_DICT and added one more test with incorrect sizes passed as kwargs, in which case the defaults from `self.min_pixels` have to be used

Ready for re-review